src/backend/optimizer.py

```diff
@@ -9,4 +14,28 @@ def __init__(self, producer):
 
     # The main function that optimizes the schedule. How the schedule and job should be implemented is up for discussion
     def optimize(self, schedule):
-        pass
+        objective = np.zeros(len(schedule))
+        return optimize.minimize(self.to_minimize, objective, tol=1.0, method="powell")
```

Comment:
Schedule is an array of jobs. `to_minimize()` is the function we want to minimize. `objective` is the initial conditions: an array of zeros. Each number in `objective` represent the job at the corresponding index in `schedule`. When optimization is finished, jobs with positive numbers are accepted, while jobs with negative numbers are rejected.